wordpress-mobile/WordPress-iOS
# Jetpack App: Fix view state for "unavailable" scan state

Fixes #16725 
Fixes #16755 

## Description

This PR fixes a bug where the "no threats" scan status was being displayed when scan state is "unavailable".

Here's what happens now If the scan state returns `unavailable`:
- If the reason is `multisite_not_supported`, we show a "WordPress multisites are not supported" message
- Otherwise, we show a generic error message "Something went wrong"

WPKit: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/415

Ref: pcdRpT-qC-p2#comment-536

Unavailable: multisites not supported | Unavailable: generic error | No threats
-- | -- | --
![Simulator Screen Shot - iPhone 12 Pro - 2021-07-07 at 11 00 06](https://user-images.githubusercontent.com/6711616/124742835-0409f200-df15-11eb-9b55-d56d3664c8a2.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-07-07 at 11 17 58](https://user-images.githubusercontent.com/6711616/124742844-0704e280-df15-11eb-9c72-1c270053e6a7.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-07-07 at 11 00 17](https://user-images.githubusercontent.com/6711616/124742852-08360f80-df15-11eb-8fc4-dfbd30385903.png)

 
## How to test
1. Go to the Jurassic Ninja special ops page
2. Select the Jetpack Complete plan
3. Launch a multisite on subdomains
4. Connect to this site and open the Jetpack app
5. Go to My Site > Scan
6. ✅ Notice an error`WordPress multisites are not supported`
7. ✅ Notice the History button is disabled

## Regression Notes
1. Potential unintended areas of impact
WordPress

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested above flow on WordPress

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
